### DISCUSSION TITLE
I think the send should split memory request and send action seperately.

### DISCUSSION THREAD

**Discussion by kipade** [mutouyun/cpp-ipc#78] [Ideas]
As the topic, according to the current implement, all the send operation will copy memory from local into shared memory, so, it would become its performance bottlenecks if the user  send/receive message very frequently.

**Comment by mutouyun** [comment]
Yes, you are right.

I know this is a big performance issue. The IPC library I wrote for the company solves the copying problem by giving users the option to construct objects directly on shared memory, but that part of the code is not open source. I'm looking for time to rewrite the open source IPC library.
